account_invoice_inter_company/tests/test_inter_company_invoice.py

```diff
@@ -69,9 +69,10 @@ def test03_confirm_invoice(self):
                           self.invoice_company_a.partner_id)
         self.assertEquals(len(invoices[0].invoice_line_ids),
                           len(self.invoice_company_a.invoice_line_ids))
-        self.assertEquals(
-            invoices[0].invoice_line_ids[0].product_id,
-            self.invoice_company_a.invoice_line_ids[0].product_id)
+        if invoices[0].company_id.company_share_product:
```

Comment:
Putting an if condition in the test denotes that the test is not well designed, as you must expect predictive results. You must assure that `company_share_product` is True in your setup, not to add this conditional.